aws/ec2/monitorInstances.go

```diff
@@ -0,0 +1,284 @@
+		Statistics: []*string{aws.String("Average"), aws.String("Maximum")},
+		Dimensions: dimensions,
+	})
+	if err != nil {
+		logger.Error("Error when fetching CPU stats from CloudWatch", err.Error())
+		return 0, 0
+	} else if len(stats.Datapoints) > 0 {
+		return aws.Float64Value(stats.Datapoints[0].Average), aws.Float64Value(stats.Datapoints[0].Maximum)
+	} else {
+		return 0, 0
+	}
+}
+
+// fetchInstancesList, firstly, calls DescribeInstances from AWS to list all the instances in a region.
+// Secondly, it generates an InstanceInfo filled by DescribeInstances, getAccountID and getInstanceStats.
+// Finally, it sends the InstanceInfo generated in a chan.
+func fetchInstancesList(ctx context.Context, creds *credentials.Credentials,
+	region string, account string, instanceInfoChan chan InstanceInfo) {
+	defer close(instanceInfoChan)
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials: creds,
+		Region:      aws.String(region),
+	}))
+	svc := ec2.New(sess)
+	instances, err := svc.DescribeInstances(nil)
+	if err != nil {
+		logger.Error("Error when describing instances", err.Error())
+		return
+	}
+	start, end := getCurrentCheckedHour()
+	for _, reservation := range instances.Reservations {
+		for _, instance := range reservation.Instances {
+			CpuAverage, CpuPeak := getInstanceStats(ctx, aws.StringValue(instance.InstanceId), sess, region)
+			instanceInfoChan <- InstanceInfo{
+				Account:    account,
+				Service:    "EC2",
+				Id:         aws.StringValue(instance.InstanceId),
+				Region:     aws.StringValue(instance.Placement.AvailabilityZone),
+				StartDate:  start,
+				EndDate:    end,
+				KeyPair:    aws.StringValue(instance.KeyName),
+				Tags:       getInstanceTag(instance.Tags),
+				Type:       aws.StringValue(instance.InstanceType),
+				CpuAverage: CpuAverage,
+				CpuPeak:    CpuPeak,
+			}
+		}
+	}
+}
+
+// fetchRegionsList fetchs the regions list from AWS and returns an array of their name.
+func fetchRegionsList(ctx context.Context, sess *session.Session) ([]string, error) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	svc := ec2.New(sess)
+	regions, err := svc.DescribeRegions(nil)
+	if err != nil {
+		logger.Error("Error when describing regions", err.Error())
+		return []string{}, err
+	}
+	res := make([]string, 0)
+	for _, region := range regions.Regions {
+		res = append(res, aws.StringValue(region.RegionName))
+	}
+	return res, nil
+}
+
+// createIndexEs creates the ElasticSearch index with the properties
+// from getPropertiesMapping.
+func createIndexEs(ctx context.Context, client *elastic.Client) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	mapping := "{\"mappings\":{\"" + DocumentEc2 + "\":" + mappingEc2Instance + "}}"
+	if _, err := client.CreateIndex(IndexHourlyInstanceUsage).BodyString(mapping).Do(context.Background()); err != nil {
+		logger.Error("Error when creating ElasticSearch index", err.Error())
```

Comment:
If your function can have an error, it should return one. How will the calling function know if this index creation has succeeded ? Should you continue the execution of your calling function if this function fails ?